### DISCUSSION TITLE
[proposal] support target specific config in moon.pkg.json

### DISCUSSION THREAD

**Discussion by bobzhang** [moonbitlang/moon#240] [Announcements]
# Feature Request

Currently we use xx.wasm.mbt to support target specific build, this is too limited and does not compose well, for example, we want to have one file for target a and b, and the other for target c, we have to create file.targeta, file.targetb, and file.targetc while file.targeta and file.targetb is the same. we would like to introduce a field in `moon.pkg.json`, for example:

```json
{
  "targets" : [
      "filea" : [":or", "wasm", "wasm-gc", "debug", "windows" ]
      "fileb" : [":and", "js" "release"]
   ]
} 
```
So for each file in the targets field, it will be excluded if the target expression does not evaluate to true

## Questions

What tags are we going to support, for some tags like debug, does it affect its dependencies

**Comment by lijunchen** [comment]
Currently, we have the following combinations: `{release, debug} x {wasm, wasm-gc, js}`. Each set can be a subset rather than just a single choice. In the future, we may have additional tags for arch, OS, and user-defined tags.

`debug` tags and others will affect downstream dependencies. For example, if B depends on A, and we are currently working on B, then A's build options will be determined by the build options set for B.

By the way, we also need to consider how to design `moondoc` to accommodate such combinations @Yoorkin
